recipes/openai_terms.py

```diff
@@ -0,0 +1,122 @@
+    # that stuff worthwhile.
+    if not path.suffix == ".jinja2":
+        msg.fail(
+            "The --prompt-path (-p) parameter expects a .jinja2 file.",
+            exits=1,
+        )
+    with path.open("r", encoding="utf8") as file_:
+        text = file_.read()
+    return jinja2.Template(text)
+
+
+def _parse_terms(completion):
+    # Sometimes it only returns a single item. For example, when there are
+    # many, many seeds around.
+    if "\n" not in completion:
+        return [completion.replace("-", "").strip().lower()]
+    # Other times we cannot assume the final item will have had sufficient 
+    # tokens available to complete the term, so we have to discard it.
+    lines = completion.split("\n")
+    return [item.replace("-", "").strip().lower() for item in lines][:-1]
+
+
+@prodigy.recipe(
+    "terms.openai.fetch",
+    query=("Query to send to OpenAI", "positional", None, str),
+    output_path=("Path to save the output", "positional", None, Path),
+    seeds=("One of more comma-seperated seed phrases.","option","s",lambda d: d.split(","),),
+    n=("Number of items to generate", "option", "n", int),
+    model=("GPT-3 model to use for completion", "option", "m", str),
+    prompt_path=("Path to jinja2 prompt template", "option", "p", Path),
+    verbose=("Print extra information to terminal", "flag", "v", bool),
+    resume=("Resume by loading in text examples from output file.", "flag", "r", bool),
+    progress=("Print progress of the recipe.", "flag", "pb", bool),
+    max_tokens=("Max tokens to generate", "option", "t", int),
+)
+def terms_openai_fetch(
+    query: str,
+    output_path: Path,
+    seeds: List[str] = [],
+    n: int = 100,
+    model: str = "text-davinci-003",
+    prompt_path: Path = DEFAULT_PROMPT_PATH,
+    verbose: bool = False,
+    resume: bool = False,
+    progress: bool = False,
+    max_tokens=100,
+):
+    """Get bulk term suggestions from an OpenAI API, using zero-shot learning.
+    The results can then be corrected using the `terms.openai.correct` recipe.
+
+    This approach lets you get the openai queries out of the way upfront, which can help
+    if you want to use multiple annotators of if you want to make sure you don't have to
+    wait on the OpenAI queries. The downside is that you can't flag examples to be integrated
+    into the prompt during the annotation.
+    """
+    template = _load_template(prompt_path)
+    terms = []
+    if resume:
+        examples = srsly.read_jsonl(output_path)
+        seeds.extend([e["text"] for e in examples])
+    headers = {
+        "Authorization": f"Bearer {os.getenv('OPENAI_KEY')}",
+        "OpenAI-Organization": os.getenv("OPENAI_ORG"),
+        "Content-Type": "application/json",
+    }
+    while len(terms) < n:
+        prompt = template.render(n=n, examples=seeds + terms, description=query)
+        if verbose:
+            rich.print(Panel(prompt, title="Prompt to OpenAI"))
+        resp = httpx.post(
```

Comment:
In case you're encountering problems with timeouts (not the 429 error), you might try using a [`httpx.Client`](https://www.python-httpx.org/advanced/) to make the `POST` call.